helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -456,15 +450,17 @@ public String toString() {
        + (_resources != null ? Arrays.toString(_resources.toArray()) : "") + "])";
   }
 
+  // TODO: to clean up, finalize is deprecated in Java 9
   @Override
   public void finalize() {
+    super.finalize();
```

Comment:
It might depend on specific logic. But I feel that we should do close() first, and than finalize()? It makes more sense to put finalize the last step.

Reply:
That makes sense, thanks.